src/Logary/Configuration_Config.fs

```diff
@@ -145,35 +142,45 @@ let runLogary conf =
 
 /// Shutdown logary, waiting maximum flushDur + shutdownDur.
 [<CompiledName "ShutdownLogary">]
-let shutdown' (flushDur : Duration) (shutdownDur : Duration)
-  ({ registry = reg; metadata = { logger = lgr } } : LogaryInstance)
-  = 
-  let log = LogLine.setPath "Logary.Configuration.Config.shutdown" >> Logger.log lgr
-  async {
-  LogLine.info "start shutdown" |> log
-  let! res = Advanced.flushAndShutdown flushDur shutdownDur reg
-  LogLine.info "stop shutdown" |> log
-  Logging.shutdownFlyweights ()
-  shutdownLogger lgr
-  return res
+let shutdown (flushDur : Duration) (shutdownDur : Duration) (inst : LogaryInstance) =
+  let log =
+    Message.setName (PointName ["Logary"; "Configuration"; "Config"; "shutdown"])
+    >> Logger.log inst.runtimeInfo.logger
+
+  job {
+    Message.info "start shutdown" |> log
+    let! res = Advanced.flushAndShutdown flushDur shutdownDur inst.registry
+    Message.info "stop shutdown" |> log
+    Logging.shutdownFlyweights ()
+    shutdownLogger inst.runtimeInfo.logger
+    return res
   }
 
 /// Shutdown logary, waiting maximum 30 seconds, 15s for flush and 15s for
 /// shutdown.
 [<CompiledName "ShutdownLogary"; Extension>]
-let shutdown =
-  shutdown' (Duration.FromSeconds 15L) (Duration.FromSeconds 15L)
+let shutdownSimple =
+  shutdown (Duration.FromSeconds 15L) (Duration.FromSeconds 15L)
 
 /// Wrap the LogaryInstance as a LogManager
 [<CompiledName "AsLogManager"; Extension>]
 let asLogManager (inst : LogaryInstance) =
-  let run = Async.RunSynchronously
+  let run = Job.Global.run
   { new LogManager with
-      member x.RuntimeInfo        = inst.metadata
-      member x.GetLogger name     = name |> getLogger inst.registry |> run
-      member x.FlushPending dur   = Advanced.flushPending dur inst.registry |> run
-      member x.Shutdown fDur sDur = shutdown' fDur sDur inst |> run
-      member x.Dispose ()         = shutdown inst |> Async.Ignore |> run
+      member x.runtimeInfo =
+        inst.runtimeInfo
+
+      member x.getLogger name =
+        name |> getLogger inst.registry |> run
+
+      member x.flushPending dur =
+        Advanced.flushPending dur inst.registry |> run
```

Comment:
On an interface level yes. On a "Logger Extensions" level, there would be another method that also ran it, which will be more common to use.

Before spending any large amounts of time on this I recommend you to fix the cyclic message buffers (overwrite oldest Message), because then this becomes mostly a non-issue.